crates/uv-toolchain/src/discovery.rs

```diff
@@ -51,25 +51,48 @@ pub enum ToolchainRequest {
     /// Generally these refer to uv-managed toolchain downloads.
     Key(PythonDownloadRequest),
 }
+
 #[derive(Debug, Default, Clone, Copy, PartialEq, Eq, serde::Deserialize)]
 #[serde(deny_unknown_fields, rename_all = "kebab-case")]
 #[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
 pub enum ToolchainPreference {
-    /// Only use managed interpreters, never use system interpreters.
+    /// Only use managed toolchains, never use system toolchains.
     OnlyManaged,
-    /// Prefer installed managed interpreters, but use system interpreters if not found.
-    /// If neither can be found, download a managed interpreter.
+    /// Prefer installed managed toolchains, over system toolchains.
+    ///
+    /// If `toolchain-fetch` is `if-necessary`, managed toolchains will only be downloaded if a
+    /// system interpreter is not found. Set `toolchain-fetch` to `always` to prefer downloading
+    /// a managed toolchain over using a system interpreter.
     #[default]
-    PreferInstalledManaged,
-    /// Prefer managed interpreters, even if one needs to be downloaded, but use system interpreters if found.
     PreferManaged,
-    /// Prefer system interpreters, only use managed interpreters if no system interpreter is found.
+    /// Prefer system toolchains, only use managed toolchains if no system interpreter is found.
+    ///
+    /// System toolchains are preferred regardless of the `toolchain-fetch` option.
     PreferSystem,
-    /// Only use system interpreters, never use managed interpreters.
+    /// Only use system toolchains, never use managed toolchains.
     OnlySystem,
 }
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, serde::Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum ToolchainFetchStrategy {
+    /// Always fetch a managed toolchain if it can be used to fulfull a request.
+    ///
+    /// If `toolchain-preference` is `only-system` this is equivalent to `never`.
+    /// If `toolchain-preference` is `prefer-system` this is equivalent to `if-necessary`.
+    Always,
```

Comment:
I worry a little bit about the interaction between these two options being confusing, but I think it'd be worse to try to enumerate in a single setting.

Reply:
Yeah hmmm. Would it help to tie them back to a specific user experience? Like, "You might want to use `only-system` because <blah>, and so setting this to `always` has no effect." Honestly not sure though.

Reply:
Idk. Maybe warnings would be sensible? Do you think they make sense as separate options?

Reply:
Like, an alternative is `toolchain-fetch-allowed` as a bool and `toolchain-preference` has `only-managed`, `only-system`, `managed`, `system`, `installed` (default), where `installed` prefers any installed toolchain over any not-yet-downloaded one  i.e. replacing `if-necessary`.

Reply:
When I was reading the docs, it did take me a minute to put the pieces together. But after having understood it, yeah, it does feel like they belong as two different options. They seem _almost_ orthogonal, but not quite.

I'd be in favor of a warning for things like `pref=only-system` and `strat=always` combinations that are probably a mistake.